### ISSUE TITLE
Cannot install: ERROR: One or more destination directories are not accessible !!!

### ISSUE THREAD

**Issue by ritog** [martin-saurer/jkernel#20]
I see this error:

```
ERROR: One or more destination directories are not accessible !!!
```

And I am unable to install J kernel on my machine.
____

Here is the full error message:

```
(j) ~/Applications/jkernel-master> python setup.py install

Check pre-requisites ...
Anaconda/Miniconda root directory ... : /home/ritog/.local OK.
Site Packages directory ............. : /home/ritog/.local/lib/python3.10/site-packages OK.
Jupyter Notebook syntax directory ... : /home/ritog/.local/lib/python3.10/site-packages/notebook/static/components/codemirror/mode MISSING or NOT writeable.
Jupyter Kernels directory ........... : /home/ritog/.local/share/jupyter/kernels MISSING or NOT writeable.
ERROR: One or more destination directories are not accessible !!!
```

How can I solve this problem?

____ 

Note: I have a proper J installation that works through the ijconsole, jconsole, and jsx (on browser).

**Comment by martin-saurer** [comment]
May I ask you a few questions?:
- Are you on a Mac?
- Which "python" are you executing on the command line? The one from MacOS or the one from Anaconda?

I recommend installing Anaconda into your home directory: /home/ritog/anaconda3
Then execute the command line: /home/ritog/anaconda3/bin/python setup.py install
setup.py builds all the necessary paths relative to the "python" executable.

Please let me know, if that helped.

**Comment by ghosh-r** [comment]
Hi @martin-saurer , I am not on Mac, but on Linux (Pop OS 22.04 LTS).

I am using miniconda, and it is indeed in `/home/ritog/miniconda3`.

I have created a conda environment, and activated it and installed jupyter in it, and I am running `python` from within that environment.

**Comment by martin-saurer** [comment]
Hi @ghosh-r ,
I have installed Miniconda, created an environment, installed jkernel, and everything worked right out-of-the-box.
I'm on Linux Mint 20.3.

`
$ conda env list
# conda environments:
#
                         /home/martin/Anaconda3
                         /home/martin/Anaconda3_NLP
                         /home/martin/Anaconda3_oTree
base                     /home/martin/Miniconda3
jj                    *  /home/martin/Miniconda3/envs/jj
`

`
$ which python
/home/martin/Miniconda3/envs/jj/bin/python
`

`
cd /home/martin/Program/jkernel
python setup.py check
Check pre-requisites ...
Anaconda/Miniconda root directory ... : /home/martin/Miniconda3/envs/jj OK.
Site Packages directory ............. : /home/martin/Miniconda3/envs/jj/lib/python3.10/site-packages OK.
Jupyter Notebook syntax directory ... : /home/martin/Miniconda3/envs/jj/lib/python3.10/site-packages/notebook/static/components/codemirror/mode OK.
Jupyter Kernels directory ........... : /home/martin/Miniconda3/envs/jj/share/jupyter/kernels OK.
ALL destination directories are OK.
Done.

Checking current installation ...
/home/martin/Miniconda3/envs/jj/lib/python3.10/site-packages/jkernel => OK
/home/martin/Miniconda3/envs/jj/lib/python3.10/site-packages/jkernel/__init__.py => OK
/home/martin/Miniconda3/envs/jj/lib/python3.10/site-packages/jkernel/jinter.py => OK
/home/martin/Miniconda3/envs/jj/lib/python3.10/site-packages/jkernel/jkernel.py => OK
/home/martin/Miniconda3/envs/jj/share/jupyter/kernels/jkernel => OK
/home/martin/Miniconda3/envs/jj/share/jupyter/kernels/jkernel/kernel.json => OK
/home/martin/Miniconda3/envs/jj/share/jupyter/kernels/jkernel/logo-32x32.png => OK
/home/martin/Miniconda3/envs/jj/share/jupyter/kernels/jkernel/logo-64x64.png => OK
/home/martin/Miniconda3/envs/jj/lib/python3.10/site-packages/notebook/static/components/codemirror/mode/j => OK
/home/martin/Miniconda3/envs/jj/lib/python3.10/site-packages/notebook/static/components/codemirror/mode/j/j.js => OK
Environment variable: J_INSTALLATION_FOLDER => OK
/home/martin/J903 => OK
Environment variable: J_BIN_FOLDER => OK
/home/martin/J903/bin => OK
ALL checks passed.
Done.
`

There must be something different in your installation, but I cannot see what it is.
It looks strange to me, that in your environment the directory .../notebook/static/components/codemirror/mode seems to be missing.

Does Jupyter Notebook (Python Kernel) work in your environment?

**Comment by ghosh-r** [comment]
> Does Jupyter Notebook (Python Kernel) work in your environment?

Yes, @martin-saurer it does. Jupyter Notebook works fine, and also I can run it with a Python3 kernel.

It might be of interest to you that after I failed to install this program, I went ahead with the J download that comes bunched up with Jupyter.

That seems to work out fine.

I downloaded the file from [J software website's Jupyter download page](http://www.jsoftware.com/download/jupyter/). And followed the instruction provided on [the Jupyter Guide page](https://code.jsoftware.com/wiki/Guides/Jupyter).
